src/sap.ui.core/src/sap/ui/model/odata/v4/lib/_Requestor.js

```diff
@@ -113,7 +113,8 @@ sap.ui.define([
 		"Accept" : "application/json;odata.metadata=minimal;IEEE754Compatible=true",
 		"OData-MaxVersion" : "4.0",
 		"OData-Version" : "4.0",
-		"X-CSRF-Token" : "Fetch"
+		"X-CSRF-Token" : "Fetch",
+		"X-Requested-With" : "XMLHttpRequest"
```

Comment:
Why should WE do this? Shouldn't it be already done automaticall by jQuery.ajax()?

Their code says the following:

				// X-Requested-With header
				// For cross-domain requests, seeing as conditions for a preflight are
				// akin to a jigsaw puzzle, we simply never set it to be sure.
				// (it can always be set on a per-request basis or even using ajaxSetup)
				// For same-domain requests, won't change header if already provided.
				if ( !options.crossDomain && !headers[ "X-Requested-With" ] ) {
					headers[ "X-Requested-With" ] = "XMLHttpRequest";
				}

Who am I to know this better? This sound dubious to me. I am not convinced, sorry.

Reply:
It would make sense that this was indeed taken care of by jQuery.ajax(), but I didn't assume that they should. When I performed some tests I quickly noticed these headers were missing using the v2/v4 OData models. 
I just did another test and it shows that these headers are present in bare $.ajax() calls, but not in ajax requests originating from UI5 v2/v4 models.

See network log: https://plnkr.co/edit/16J1TFICxbqETCzaxuZ0?preview
To whitelist your IP for the V2 CORS, click the button over at https://cors-anywhere.herokuapp.com/corsdemo

Reply:
Have you shared the correct Plunker ID? It is not doing any OData calls in my case and in the code I also cannot see why it should.
With one of our test applications I can see the header `X-Requested-With: XMLHttpRequest` for the batch requests.

Reply:
@uhlmannm The Plunker issues a same-origin request for its JSONModel data. I guess that's what @piejanssens was referring to.

As @ThomasChadzelek already pointed out with the code snippet, jQuery sends the `X-Request-With=XMLHttpRequest` header automatically, if and only if the outgoing request is a same-origin request and when the caller of `$.ajax` did not add the header already.

The reason for the same-origin condition is that `X-Requested-With` is not a [CORS-safelisted header](https://fetch.spec.whatwg.org/#cors-safelisted-request-header). Adding it to a cross-origin requests will therefore trigger an additional preflight request, which often is unwanted in the general case (e.g. for performance reasons or when the server is not prepared to handle it). Several frameworks therefore removed the `X-Requested-With` header again from their defaults (1) or limited it to the same-origin case (jQuery).

When there are already other non-safelisted headers (e.g. X-CSRF-Token) and when the server is known to allow the `X-Requested-With` header or when the server is in the same-origin as the current page, then adding the header should be safe.

I just wonder how the `ODataModel` could know about the server capabilities. 

[1] https://github.com/ReactiveX/rxjs/issues/3273 , https://github.com/angular/angular.js/issues/1004